src/native-stack/views/NativeStackView.tsx

```diff
@@ -78,6 +130,24 @@ export default function NativeStackView({
           contentStyle,
         ];
 
+        if (
+          !didWarn &&
+          stackPresentation !== 'push' &&
+          headerShown !== undefined
+        ) {
+          didWarn = true;
+          console.warn(
+            'Be aware that changing the visibility of header in modal on iOS will result in resetting the state of the screen.'
+          );
+        }
```

Comment:
@WoLewicki Would it not make more sense to warn about this whenever a visibility change actually happens?

Now I get a warning every time I run the app and open a modal despite having a static `headerShown: false` that will never cause this issue.

Reply:
We would need to keep track of the previous value of this prop in order to do that, and I am not sure if it is a good idea. Also, if you do not want the header to be shown, you can just not change the prop in modal screens at all. You can easily remove the warning by using `LogBox.ignoreLogs`.

Reply:
Yeah, good point.

I see it's triggered due to `screenOptions={{ headerShown: false }}` on the navigator. I know you suggest setting `headerShown` on each screen instead, but why is that suggested exactly?

>You can easily remove the warning by using LogBox.ignoreLogs.

`headerShown: undefined` on the modal screen works as well, as it seems to override the navigator's `headerShown`.

Reply:
Yeah, setting props on individual screens override the options set in navigator's `screenOptions`.

Reply:
I understand, then basically I can ignore the warning if I did not intentionally change `headerShown` prop. Just to make sure I am not doing anything wrong by paying attention to all warnings. :)